api/notd/store/retriever.py

```diff
@@ -118,3 +119,25 @@ async def get_collection_by_address(self, address: str, connection: Optional[Dat
             raise NotFoundException(message=f'Collection with registry {address} not found')
         collection = collection_from_row(row)
         return collection
+
+    async def get_collection_holding(self, address: str, ownerAddress: str, connection: Optional[DatabaseConnection] = None) -> List[Token]:
+        boughtTokens = []
+        soldTokens= []
+        query = TokenTransfersTable.select()\
+            .where(TokenTransfersTable.c.registryAddress == address)
+        query = query.where(TokenTransfersTable.c.toAddress == ownerAddress)
+        for row in await self.database.execute(query=query, connection=connection):
+            tokenTransfer = token_transfer_from_row(row)
+            boughtTokens.append(tokenTransfer)
+        query = TokenTransfersTable.select()
+        query = query.where(TokenTransfersTable.c.registryAddress == address)
+        query = query.where(TokenTransfersTable.c.fromAddress == ownerAddress)
+        for row in await self.database.execute(query=query, connection=connection):
+            tokenTransfer = token_transfer_from_row(row)
+            soldTokens.append(tokenTransfer)
```

Comment:
```suggestion
        result = await self.database.execute(query=query, connection=connection)
        soldTokens = [token_transfer_from_row(row) for row in result]
```

Reply:
you should get out of the habit of awaiting and then doing something with the result in one operation. try to always await and then do stuff on the next line